### DISCUSSION TITLE
Webpack5 Angular SyntaxError: HookWebpackError: Module build failed

### DISCUSSION THREAD

**Discussion by g-gibosky** [webpack/webpack#18712] [Q&A]
# Bug report

I'm migrating a Angular v4 application to v18 and I'm having a lot of problem with webpack builds.

I migrated to angular cli but the default one didn't handle everything that i need for the project.

The command ng serve return this error:

Build at: 2024-08-27T09:12:34.969Z - Hash: c0e3cf9439d487fc - Time: 20039ms

./src/assets/css/styles.css?ngGlobalStyle - Error: Module build failed (from ./node_modules/@angular-devkit/build-angular/node_modules/mini-css-extract-plugin/dist/loader.js): HookWebpackError: Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):

SyntaxError

(2:7) /home/librarian/Documents/static/src/assets/css/styles.css Unknown word

import API from "!../../../node_modules/style-loader/dist/runtime/injectStylesIntoStyleTag.js"; import domAPI from "!../../../node_modules/style-loader/dist/runtime/styleDomAPI.js"; import insertFn from "!../../../node_modules/style-loader/dist/runtime/insertBySelector.js";

**What is the current behavior?**

Can't build application

Angular 18 with handsontable external resource and with custom build.

Configuration files are here:

[https://stackoverflow.com/questions/78918299/webpack5-angular-syntaxerror-hookwebpackerror-module-build-failed](url)

**What is the expected behavior?**

Build correctly

**Other relevant information:**
webpack version:  
Node.js version: 18.20
Operating System: Windows
Additional tools: Vscode

**Comment by alexander-akait** [comment]
Looks like you have custim webpack configuration and have wrong regexp, because you try to handle JS using CSS parser, please provide reprducible test repo or open an issue in angular ng repo, thank you

**Comment by g-gibosky** [comment]
Thanks for the reply.

Here is the config files.

angular.json:

```
{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "static": {
      "projectType": "application",
      "schematics": {},
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-builders/custom-webpack:browser",
          "options": {
            "outputPath": "dist/dev",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "assets": [
              {
                "glob": "**/*",
                "input": "public",
                "output": "/assets"
              }
            ],
            "styles": [
              "src/assets/css/styles.css"
            ],
            "scripts": [],
            "vendorChunk": true,
            "extractLicenses": true,
            "buildOptimizer": true,
            "optimization": true,
            "sourceMap": false,
            "customWebpackConfig": {
              "path": "./webpack/webpack.common.js"
            }
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kB",
                  "maximumError": "1MB"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kB",
                  "maximumError": "4kB"
                }
              ],
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "outputHashing": "all",
              "sourceMap": false,
              "optimization": true,
              "customWebpackConfig": {
                "path": "./webpack/webpack.prod.js"
              }
            },
            "development": {
              "optimization": false,
              "sourceMap": true,
              "extractLicenses": false,
              "customWebpackConfig": {
                "path": "./webpack/webpack.dev.js"
              }
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-builders/custom-webpack:dev-server",
          "options": {
            "buildTarget": "static:build"
          },
          "configurations": {
            "production": {
              "buildTarget": "static:build:production"
            },
            "development": {
              "buildTarget": "static:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n"
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "assets": [
              {
                "glob": "**/*",
                "input": "public",
                "output": "/assets"
              }
            ],
            "styles": [
              "src/assets/css/styles.css",
              "../node_modules/bootstrap/dist/css/bootstrap.css"
            ],
            "scripts": []
          }
        }
      }
    }
  }
}
```

webpack.common.js:

```
const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');

module.exports = {
  plugins: [
    new HtmlWebpackPlugin({
      template: './src/index.html',
      inject: true,
      title: 'New Project',
      minify: {
        collapseWhitespace: true,
        minifyCSS: true,
        minifyJS: true,
        preserveLineBreaks: true
      }
    })
  ],
  entry: {
    app: './src/main.ts'
  },
  resolve: {
    extensions: ['.js', '.ts'],
    alias: {
      handsontable: path.resolve(__dirname, 'src/handsontable/dist/handsontable.full.js'),
    }
  },
  module: {
    rules: [
      {
        test: /\.ts$/,
        use: [
          {
            loader: 'ts-loader',
            options: { transpileOnly: true }
          },
        ],
        exclude: /node_modules/
      },
      {
        test: /\.(png|jpe?g|gif|svg|woff|woff2|ttf|otf|eot|ico)$/,
        type: 'asset/resource',
      },
      {
        test: /\.css$/,
        include: /src\/app/,
        use: ['raw-loader']
      }
    ]
  }
};
````

webpack.dev.js:

```
const { merge } = require('webpack-merge');
const common = require('./webpack.common.js');
const path = require('path');

module.exports = merge(common, {
  mode: 'development',
  devtool: 'inline-source-map',
  devServer: {
    static: {
      directory: path.join(__dirname, 'dist'),
    },
    open: true,
    compress: true,
    port: 4200
  },
  module: {
    rules: [
      {
        test: /\.css$/,
        exclude: /src\/app/,
        use: ['style-loader', 'css-loader', 'postcss-loader'],
      },
    ]
  },
});
```

Also there is my project directory structure:

├── handsontable
│   ├── dist copy
│   │   ├── languages
│   │   ├── handsontable.css
│   │   ├── handsontable.full.css
│   │   ├── handsontable.full.js
│   │   ├── handsontable.full.min.css
│   │   ├── handsontable.full.min.js
│   │   ├── handsontable.js
│   │   ├── handsontable.min.css
│   │   ├── handsontable.min.js
│   │   └── README.md
│   ├── utils
│   │   ├── constants.ts
│   │   └── hooks-callbacks.ts
│   ├── custom_handsontable.css
│   ├── handsontable.component.html
│   ├── handsontable.component.ts
│   └── handsontable.module.ts
├── public
│   └── favicon.ico
├── src
│   ├── app
│   │   ├── allocations
│   │   ├── calendar
│   │   ├── export
│   │   ├── fonts
│   │   ├── programs
│   │   ├── shared
│   │   ├── teams
│   │   ├── validation
│   │   ├── app.component.ts
│   │   ├── app.module.ts
│   │   └── app.routing.ts
│   ├── assets
│   │   ├── css
│   │   └── img
│   ├── environments
│   │   ├── environment.prod.ts
│   │   └── environment.ts
│   ├── index.html
│   ├── main.ts
│   ├── polyfills.ts
│   ├── services.ts
│   ├── utils.ts
│   └── vendor.ts
├── typings
│   ├── globals
│   │   ├── core-js
│   │   ├── jasmine
│   │   └── node
│   └── index.d.ts
├── webpack
│   ├── helpers.js
│   ├── webpack.common.js
│   ├── webpack.dev.js
│   └── webpack.prod.js
├── angular.json
├── ispinner.css
├── lodash.custom.js
├── lodash.custom.min.js
├── model.ts
├── package.json
├── package-lock.json
├── package-lock.json.bkp
├── postcss.config.js
├── README.md
├── server.ts
├── tsconfig.app.json
├── tsconfig.json
├── tsconfig.spec.json
├── typings.json
└── webpack.config.js.bkp

**Comment by alexander-akait** [comment]
I think angular has own CSS loaders and configuration for them, that is why they are have overlapping, I think you can ask this question in angular repo (`angular-cli`)
